src/platform_impl/linux/kms/mod.rs

```diff
@@ -0,0 +1,166 @@
+use crate::dpi::{PhysicalPosition, PhysicalSize};
+
+pub mod event_loop;
+pub mod input;
+pub mod window;
+use crate::platform_impl;
+pub use drm::SystemError;
+use drm::{
+    control::{Device as ControlDevice, *},
+    Device,
+};
+pub use event_loop::EventLoop;
+pub use event_loop::EventLoopProxy;
+pub use event_loop::EventLoopWindowTarget;
+use parking_lot::Mutex;
+use std::os::unix;
+use std::os::unix::prelude::FromRawFd;
+use std::sync::Arc;
+pub use window::Window;
+
+pub static MODE: Mutex<Option<Mode>> = parking_lot::const_mutex(None);
+
+#[derive(Debug, Clone)]
+/// A simple wrapper for a device node.
+pub struct Card(pub(crate) Arc<i32>);
+
+/// Implementing `AsRawFd` is a prerequisite to implementing the traits found
+/// in this crate. Here, we are just calling `as_raw_fd()` on the inner File.
+impl unix::io::AsRawFd for Card {
+    fn as_raw_fd(&self) -> unix::io::RawFd {
+        *self.0
+    }
+}
+
+impl Drop for Card {
+    fn drop(&mut self) {
+        unsafe { std::fs::File::from_raw_fd(*self.0) };
+    }
+}
```

Comment:
This doesn't look sound to me. Since the `i32` can still be used outside of the `Card` (thanks to `AsRawFd`), couldn't it still be used after a free? It might be better to define `Card` as a wrapper around an `OwnedFd` and then wrap that in an `Arc`.

Reply:
Yeah, this isn't right because `drop` is closing this every time you drop a `Card`, even if you cloned it. The dropping needs to be within the `Arc`.

Winit currently requires Rust 1.64.0, so it be correct and require less code to use `struct Card(Arc<OwnedFd>)`.